javascript/ql/src/semmle/javascript/security/dataflow/UnsafeShellCommandConstructionCustomizations.qll

```diff
@@ -199,8 +200,10 @@ module UnsafeShellCommandConstruction {
     override EqualityTest astNode;
 
     TypeOfSanitizer() {
-      exists(StringLiteral str, TypeofExpr typeof | astNode.hasOperands(str, typeof) |
-        str.getValue() = ["number", "boolean"] and // "undefined" is already handled in TaintTracking.qll
+      exists(Expr str, TypeofExpr typeof | astNode.hasOperands(str, typeof) |
+        str.mayHaveStringValue(any(InferredType t |
+            t = TTUndefined() or t = TTNumber() or t = TTBoolean()
```

Comment:
`TTUndefined` causes this to be out of sync with the qldoc above - I will let you decide on the solution.

Reply:
I removed it from the list. 
`TaintTracking::TypeOfUndefinedSanitizer` takes care of that case.